docs(style guide): add localization and internationalization guide (#1984)
asyncapi/community: docs/styleguide/localization.md

---
title: Internationalization (i18n) and Localization (l10n)
description: A guide to translating AsyncAPI documentation into multiple languages
weight: 170
---

## Internationalization and Localization

> As AsyncAPI gains more users worldwide, it becomes crucial to create tutorials, guides, and other documentation that are accessible and adaptable to different cultures. Internationalization (i18n) is the process of designing and preparing content so it can be easily adapted to multiple languages and locales. Localization (l10n), on the other hand, is the actual process of adapting that content or software for a specific language, region, or culture. This guide provides recommendations and tips to help contributors achieve this goal.

Reviewer: > to help contributors achieve this goal.

It is unclear what 'this' refers to in this context. Localization? Internationalization? Or both?

Author: Both

## Key Principles of Internationalization

> When designing AsyncAPI's documentation to be adaptable to different languages, apply the following principles:

Reviewer: Basically, it's repetition of what you've already said above
 
```suggestion
```

### Language Considerations

* Write content in clear, simple English that can be easily translated
* Avoid idiomatic expressions, colloquialisms, and region-specific references
* Use straightforward sentence structures

> * Use complex grammatical constructions only when necessary

Reviewer: I would suggest rephrasing it a bit to simplify comprehension. You start with 'use' but the main idea reveals only at the end of the sentence: 'where possible". Basically, what you're trying to say is to "Avoid complex constructions where possible"

```suggestion
* Avoid complex grammatical constructions where possible
```

### Cultural Sensitivity

* Avoid using cultural references and humor that may not translate or be understood globally
* Be mindful of cultural nuances when writing about specific topics

> * Use neutral, professional language that resonates across different cultures

Reviewer: ```suggestion
* Use neutral, commonly understood language that appeals to a diverse audience
```

## Localization Best Practices

> When adapting AsyncAPI's content to a specific language and its corresponding audience, consider doing the following:

Reviewer: ```suggestion
```

### Translation Preparation

* Use consistent terminology throughout documentation
* Create a centralized glossary of technical terms
* Provide context for translators
* Leave space for translated content to be added later

> ### Structural Considerations

Reviewer: What about currencies?

* Ensure documentation layout supports right-to-left (RTL) and left-to-right (LTR) languages
* Use flexible design that accommodates varying text lengths

> * Use dates, units of measurement, and time formats that are consistent across languages

Reviewer: I believe it should be defined more precisely because it's kinda sensitive area in terms of how one reads dates and time.
